components/web_discovery/browser/regex_util.cc

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/web_discovery/browser/regex_util.h"
+
+#include "base/strings/strcat.h"
+#include "base/strings/string_number_conversions.h"
+
+namespace {
+
+constexpr char kLongNumberRegexPrefix[] = "[0-9]{";
+constexpr char kLongNumberRegexSuffix[] = ",}";
+constexpr char kEmailRegex[] =
+    "[a-z0-9\\-_@]+(@|%40|%(25)+40)[a-z0-9\\-_]+\\.[a-z0-9\\-_]";
+constexpr char kHttpPasswordRegex[] = "[^:]+:[^@]+@";
+
+constexpr std::array<std::string_view, 10> kPathAndQueryStringCheckRegexes = {
+    "(?i)\\/admin([\\/\\?#=]|$)",
```

Comment:
We should probably also exclude:
- `/cpanel/`
- `/owa/` (Outlook Web Access)
- `/phpmyadmin/` (case may vary)
- `/phpinfo` or `/phpinfo.php`
- `/xmlrpc.php`